Source/Common/FEXServerClient.cpp

```diff
@@ -100,41 +100,56 @@ namespace FEXServerClient {
     return GetServerLockFolder() + "RootFS.lock";
   }
 
-  std::string GetServerSocketFile() {
-    FEX_CONFIG_OPT(ServerSocketPath, SERVERSOCKETPATH);
-    if (ServerSocketPath().empty()) {
-      auto TmpDir = std::filesystem::temp_directory_path().string();
-
-      auto XDGRuntimeEnv = getenv("XDG_RUNTIME_DIR");
-      if (XDGRuntimeEnv) {
-        // If the XDG runtime directory works then use that instead.
-        TmpDir = XDGRuntimeEnv;
-      }
+  std::string GetServerTempFolder() {
+    // We need a server temporary folder that has the following requirements.
+    // - Can't be `/tmp/`
```

Comment:
If it can't be that, why do we proceed to call `temp_directory_path`? IIUC the comment line only applies to FEXServer running as a systemd service, but that's not evident from the comment.

Reply:
Cleared up the comment a bit and changed around the ordering of the code to more correctly reflect the fact that `/tmp/` is only used as a fallback path.